platform-docs/docs/intro.md

```diff
@@ -10,76 +10,105 @@ Let's discover how to use the **Gutenberg Block Editor** to build your own block
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  // highlight-next-line
+  define: { 'process.env.IS_GUTENBERG_PLUGIN': false },
+  plugins: [react()],
+})
+```
+
 ## JSX
 
-We're going to be using JSX to write our UI and components. So one of the first steps we need to do is to configure our build tooling, By default vite supports JSX and and outputs the result as a React pragma. The Block editor uses React so there's no need to configure anything here but if you're using a different bundler/build tool, make sure the JSX transpilation is setup properly.
+We're going to be using JSX to write our UI and components as the block editor is built with React. Using the Vite bootstrap described above there’s no need to configure anything as it outputs the result as a React pragma. If you're using a different bundler/build tool, you may need to configure the JSX transpilation to do the same.
 
 ## Bootstrap your block editor
 
-It's time to render our first block editor.
+It's time to render our first block editor. Update your `src/App.jsx` file with the following code:
 
- - Update your `index.jsx` file with the following code:
 ```jsx
-import { createElement, useState } from "react";
-import { createRoot } from 'react-dom/client';
+import { useState } from "react";
 import {
   BlockEditorProvider,
   BlockCanvas,
 } from "@wordpress/block-editor";
 import { registerCoreBlocks } from "@wordpress/block-library";
+import { getBlockTypes } from "@wordpress/blocks";
 
 // Default styles that are needed for the editor.
-import "@wordpress/components/build-style/style.css";
-import "@wordpress/block-editor/build-style/style.css";
+import componentsStyles from "@wordpress/components/build-style/style.css?raw";
+import blockEditorStyles from "@wordpress/block-editor/build-style/style.css?raw";
+import blockEditorContentStyles from "@wordpress/block-editor/build-style/content.css?raw";
 
 // Default styles that are needed for the core blocks.
-import "@wordpress/block-library/build-style/common.css";
-import "@wordpress/block-library/build-style/style.css";
-import "@wordpress/block-library/build-style/editor.css";
-
-// Register the default core block types.
-registerCoreBlocks();
-
-function Editor() {
+import blocksCommonStyles from "@wordpress/block-library/build-style/common.css?raw";
+import blocksStyles from "@wordpress/block-library/build-style/style.css?raw";
+import blocksEditorStyles from "@wordpress/block-library/build-style/editor.css?raw";
+
+const styles = `
+  ${ componentsStyles }
+  ${ blockEditorStyles }
+  ${ blockEditorContentStyles }
+  ${ blocksCommonStyles }
+  ${ blocksStyles }
+  ${ blocksEditorStyles }
+`;
+
+const registeredBlockTypes = getBlockTypes();
+// Registers the default core block types, avoiding doing so again during HMR.
+if (
+  ! registeredBlockTypes.length ||
+  ! registeredBlockTypes.some((blockType) => blockType.name.startsWith('core/'))
+) registerCoreBlocks();
```

Comment:
Is there a simpler way to do this. We're trying to teach people how to build a block editor, we should try to keep things as simple as possible. For instance is there something like `! isHMR && registerCoreBlocks()` instead of using the block package.

Reply:
I didn’t see a simple way to branch for HMR but could have missed something. Reregistering the blocks was not any fatal issue but it polluted the console quite a bit. Now that I’ve tested with the latest `@wordpress` packages I’m not seeing console messages output for blocks and I don’t know why. Still, there is one that gets output for the `core/footnote` format. The simplest way I thought of to avoid this is instructing that the `registerCoreBlocks` call be made in `src/main.jsx`.